### PR TITLE
Simplify paths

### LINKED ISSUE #28: Simplify os-specific pathing
We can simplify a LOT of their directory-searching code using pathlib. It'll make things much cleaner. Planning on doing this after I finish linting.

### PR DESCRIPTION
**Pull request by arades79**
### Summary

This PR moves all pathing and file logic over to using pathlib. 

There were also significant refactorings done to the common module to simplify bootstrapping and overall simplify logic to be drastically more ergonomic to use as well as easier to maintain.

This PR fixes #28 
### Checklist

- [x] If code changes were made then they have been tested.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: hyperdome/common/common.py [hunk 1]
```diff
@@ -18,71 +18,244 @@
...
+
+@bootstrap
+def data_path() -> Path:
     """
-    Returns the absolute path of a resource, regardless of whether
-    hyperdome is installed systemwide, and whether regardless of platform_str
+    Returns the path of the hyperdome data directory.
+    """
+    home = Path.home()
+    if (appdata := (home / "AppData" / "Roaming")).exists():
```
Comment (stevenpitts) [inline hyperdome/common/common.py:49]: Walrus :D

**Review by stevenpitts — APPROVED** [review]
Love it

File: hyperdome/client/scripts/start_client.py [hunk 2]
```diff
@@ -19,15 +19,19 @@
...
 import sys
 
+import click
+
+from ...common.common import version
+
 
 @click.command()
 @click.option("--debug", "-d", is_flag=True)
+@click.version_option(version, prog_name="Hyperdome Server")
```
Comment (arades79) [inline hyperdome/client/scripts/start_client.py:31]: oops, didn't change that to client

File: hyperdome/common/common.py (hunk 1 above)
Comment (arades79) [reply hyperdome/common/common.py:49]: :=

File: hyperdome/common/onion.py [hunk 3]
```diff
@@ -174,7 +175,7 @@ def __init__(self, common):
             self.tor_geo_ip_file_path,
             self.tor_geo_ipv6_file_path,
             self.obfs4proxy_file_path,
-        ) = self.common.get_tor_paths()
+        ) = tor_paths
```
Comment (arades79) [inline hyperdome/common/onion.py:178]: I bet if I changed tor_paths to a namedTuple it would make these reassignments unecessary

**Pull request merged**